python/pyiceberg/catalog/__init__.py

```diff
@@ -75,6 +75,7 @@ class CatalogType(Enum):
     HIVE = "hive"
     GLUE = "glue"
     DYNAMODB = "dynamodb"
+    JDBC = "jdbc"
```

Comment:
Do we want to call this JDBC or do we want to call it just a "database" catalog? JDBC signals Java to me. I know it's the name we've used in the JVM implementation, but I think we should consider naming it differently here to avoid confusion. We probably can't change it later.

Reply:
I felt the same when I wrote this line of code, but wasn't sure if I had the rights to rebrand this!  `sql`, `database`, `rdbms`...?  My only concern is will that be confusing for users using both Java and Pyiceberg?  Shouldn't be, but the components will be named differently between Java and Python.  So it's a question of consistency VS accuracy.